File: src/main/java/systems/conduit/main/plugin/config/annotation/ConfigFile.java
package systems.conduit.main.plugin.config.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * @author Innectic
 * @since 10/17/2019
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
public @interface ConfigFile {
    String name();
    String type();

    String defaultFile() default "";
}

File: src/main/kotlin/systems/conduit/core/plugin/config/annotation/ConfigFile.kt
package systems.conduit.core.plugin.config.annotation

/*
 * @author Innectic
 * @since 10/17/2019
 */
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS)
annotation class ConfigFile(val name: String, val type: String, val defaultFile: String = "") 